extract_token_from_browser.py

```diff
@@ -1,64 +1,176 @@
 #!/usr/bin/env python3
+# Examples:
+# python ./extract_token_from_browser.py chrome --profile="Profile 2" \
+#   --token=xoxc-foo-bar
+# python ./extract_token_from_browser.py firefox-snap --token=xoxc-baz
 
 import argparse
 import json
-from pathlib import Path
+import os
+import secretstorage
 import sqlite3
 import sys
 
+from pathlib import Path
+from Crypto.Cipher import AES
+from Crypto.Protocol.KDF import PBKDF2
+
+
+class AESCipher:
+
+    def __init__(self, key):
+        self.key = key
+
+    def decrypt(self, text):
+        cipher = AES.new(self.key, AES.MODE_CBC, IV=(' ' * 16))
+        return self._unpad(cipher.decrypt(text))
+    
+    def _unpad(self, s):
+        return s[:-ord(s[len(s) - 1:])]
+
+
 parser = argparse.ArgumentParser(
     description="Extract Slack tokens from the browser files"
 )
 parser.add_argument(
     "browser", help="Which browser to extract from", metavar="<browser>"
 )
+parser.add_argument(
+    "--token", help="Extracted token", metavar="<token>", nargs='?'
+)
+parser.add_argument(
+    "--profile", help="Profile to look up cookies for", metavar="<profile>", nargs='?'
+)
 args = parser.parse_args()
 
-if args.browser != "firefox":
-    print("Currently only firefox is supported by this script", file=sys.stderr)
+if args.browser not in ["firefox", "firefox-snap", "chrome", "chrome-beta"]:
+    print("Currently only firefox. firefox-snap, chrome, "
+          "chrome-beta are supported by this script", file=sys.stderr)
     sys.exit(1)
 
 if sys.platform.startswith("linux"):
-    firefox_path = Path.home().joinpath(".mozilla/firefox")
+    iterations = 1
+    if args.browser == "firefox-snap":
+        browser_data = Path.home().joinpath(
+            "snap/firefox/common/.mozilla/firefox")
+    elif args.browser == "firefox":
+        browser_data = Path.home().joinpath(".mozilla/firefox")
+    else:
+        browser_data = Path.home().joinpath(".config/google-%s" % args.browser)
```

Comment:
Can also add support for chromium with the path `.config/chromium`.

Reply:
I havent' planned this PR for Chromium support, could be done as follow-ups if anyone willing to catch up?